.github/workflows/ci.yml

```
@@ -0,0 +1,26 @@
+---
+name: build
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  ci:
+    name: Run checks and tests over ${{matrix.otp_vsn}} and ${{matrix.os}}
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        otp_vsn: [19.0, 19.3, 20.0, 20.1.7, 20.3.8.22, 21.0.9, 21.1.4, 21.2.7, 21.3.8.1, 22.0.7, 22.1.8.1, 22.2.8, 22.3.4, 23.0.2]
+        os: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v2
+      - run: kerl list installations
+      - run: sudo pip install -q httpbin
+      - run: sudo pip install -q gunicorn
+      - run: "gunicorn -b 127.0.0.1:8000 -b unix:httpbin.sock httpbin:app&"
+      - run: rebar3 xref
+      - run: rebar3 eunit
+      - run: rebar3 dialyzer
```

You can use multline operations with you want to.

```suggestion
      - run: |
        kerl list installations
        sudo pip install -q httpbin
        sudo pip install -q gunicorn
        "gunicorn -b 127.0.0.1:8000 -b unix:httpbin.sock httpbin:app&"
        rebar3 xref
        rebar3 eunit
        rebar3 dialyzer
```

---

Sure, but doesn't that make it a single task? I prefer separate tasks to easily identify where exceptions occur. Is there a "standard way of doing things"?

---

If the maintainers prefer it, though, I'm all for changing it.

---

We could also do `rebar3 do xref, eunit, dialyzer`, for example, but it's separated for the same reason as before. On the other hand, if it one day becomes possible to skip an action, if it's failing, visibility is increased even further.

---

Yeah, true. Just if maintainers prefer it <3

---

In any case, the suggestion is missing whitespace :)

---

separate tasks makes maintenance easier imo :)